tests/test_unit.py

```diff
@@ -462,6 +462,41 @@ def test_related_object_entity_optimization(self):
         result = api.shotgun._translate_filters(filters, "all")
         self.assertEqual(result, expected)
 
+    def test_related_object_entity_optimization_in(self):
+        filters = [
+            [
+                "project",
+                "in",
+                [
+                    {"foo1": "foo1", "bar1": "bar1", "id": 999, "baz1": "baz1", "type": "Anything"},
+                    {"foo2": "foo2", "bar2": "bar2", "id": 998, "baz2": "baz2", "type": "Anything"}
+                ],
+            ],
+        ]
+        expected = {
+            "logical_operator": "and",
+            "conditions": [
+                {
+                    "path": "project",
+                    "relation": "in",
+                    "values": [
+                        {
+                            "id": 999,
+                            "type": "Anything",
+                        },
+                        {
+                            "id": 998,
+                            "type": "Anything",
+                        }
+                    ],
+                }
+            ],
+        }
+        os.environ["SHOTGUN_API_ENABLE_ENTITY_OPTIMIZATION"] = "1"
```

Comment:
For this kind of test, in order to leave in the same state, I prefer to use a context:
```python
from unittest import mock
with mock.patch.dict("os.environ", {
  "SHOTGUN_API_ENABLE_ENTITY_OPTIMIZATION": "1",
}):
```

Reply:
I've found a way using a decorator for the test case.